src/foam/util/SafetyUtil.java

```diff
@@ -62,9 +62,7 @@ public static int compare(Object o1, Object o2) {
     }
 
     // Handle array comparison
-    if ( o1.getClass().isArray() && o2.getClass().isArray()
-      && o1.getClass() == o2.getClass()
-    ) {
+    if ( o1.getClass().isArray() && o1.getClass() == o2.getClass() ) {
```

Comment:
I think we should still have this check, but it shouldn't be restricted to just arrays. If the classes aren't the same then compare the classes.

Reply:
Fixed in https://github.com/kgrgreer/foam3/pull/1683/commits/0d49b776535feae9800460e3f1e22f6836ebcabb.